server/src/test/java/org/elasticsearch/search/fetch/subphase/FieldFetcherTests.java

```diff
@@ -669,6 +669,64 @@ public void testNestedFields() throws IOException {
         assertEquals("value4b", eval("inner_nested.0.f4.0", obj1));
     }
 
+    @SuppressWarnings("unchecked")
+    public void testFlattenedField() throws IOException {
+        XContentBuilder mapping = XContentFactory.jsonBuilder().startObject()
+        .startObject("_doc")
+            .startObject("properties")
+              .startObject("flat")
+                .field("type", "flattened")
+              .endObject()
+            .endObject()
+        .endObject()
+        .endObject();
+
```

Comment:
You can use the `mapping` or `fieldMapping` static methods to avoid a lot of this ceremony around json builders